src/Streamly/Internal/Data/Array/PrimInclude.hs

```diff
@@ -170,8 +171,8 @@ last arr = readIndex arr (length arr - 1)
 concat :: (MonadIO m, Prim a) => SerialT m (Array a) -> SerialT m a
 -- concat m = D.fromStreamD $ A.flattenArrays (D.toStreamD m)
 -- concat m = D.fromStreamD $ D.concatMap A.toStreamD (D.toStreamD m)
-concat (SerialT m) =
-    SerialT $ D.toStreamK $ D.unfoldMany read (D.fromStreamK m)
+concat m =
+    Stream.fromStreamK $ D.toStreamK $ D.unfoldMany read (D.fromStreamK $ Stream.toStreamK m)
```

Comment:
eta reduction? `$` inside `(` `)` looks hard to read. You might as well add another `(` `)` instead.